### DISCUSSION TITLE
Parallel execution for single Spec file tests

### DISCUSSION THREAD

**Discussion by bajipogadadanda** [webdriverio/codemod#98] [General]
Hi Chris,
I have tried the parallel execution  for both single and multiple capabilities. Its perfectly working for multiple spec files!

However, our Spec files are large and would  have up to 50 tests per spec file, so is there anyway to have parallel execution for the tests within the spec file? i.e we group/execute a single spec file at a time , looking for if WDIO supports  to have multiple threads for the single spec file to execute the tests in parallel for the concerned spec file.

**Comment by christian-bromann** [comment]
> so is there anyway to have parallel execution for the tests within the spec file?

No, there is no straightforward way to do this. You should split up your test file into multiple files.
